### PR TITLE
Clear tick natives after being disconnected

### PR DESCRIPTION
**Pull request by DorCoMaNdO**
Clear tick natives after being disconnected (I noticed that the code doesn't reach it only after the fact, but it doesn't harm to have it)
Added locks to Opponents, Npcs, _vehMods, _pedClothes, _threadJumping and _tickNatives.
Fixed CheckPlayerProps() returning _vehMods.
Fixed pressing the menu key opening the main menu, even when in a submenu.

### CODE REVIEW

File: Client/Main.cs [hunk 1]
```diff
@@ -567,21 +551,7 @@ public void OnTick(object sender, EventArgs e)
...
-                _mainMenu.MenuItems[5].Text = "Disconnect";
-                _mainMenu.MenuItems[6].Enabled = true;
-                _settingsMenu.MenuItems[0].Enabled = false;
-            }
-            else
-            {
-                _mainMenu.MenuItems[5].Text = "Connect";
-                _mainMenu.MenuItems[6].Enabled = false;
-                _settingsMenu.MenuItems[0].Enabled = true;
-            }
```
Comment (Guad) [inline Client/Main.cs]: Why did you remove this?

File: Client/Main.cs [hunk 2]
```diff
@@ -694,9 +674,10 @@ public void ConnectToServer(string ip, int port = 0)
                 _client = new NetClient(_config);
                 _client.Start();
             }
-            Opponents = new Dictionary<long, SyncPed>();
-            Npcs = new Dictionary<string, SyncPed>();
-            _tickNatives = new Dictionary<string, NativeData>();
+
+            lock (Opponents) Opponents = new Dictionary<long, SyncPed>();
+            lock (Npcs) Npcs = new Dictionary<string, SyncPed>();
+            lock (_tickNatives) _tickNatives = new Dictionary<string, NativeData>();
```
Comment (Guad) [inline Client/Main.cs]: This will cause an ArgumentNullException if any of these is null

**Comment by DorCoMaNdO** [comment]
Ahh forgot one undo for the menu thing, I was going to make it so it's placed at the bottom, and the native ticks would then run from within the statement, but I reverted that, should have probably paid more attention.

And yes it will throw an exception, my bad, fix that by initializing the dictionaries from their definition, that part has to have locks, it executes whenever a player connects, and if any of the values are being used for whatever reason elsewhere in the code, it will fail.

**Comment by Guad** [comment]
Yeah I ended up doing that.

**Comment by DorCoMaNdO** [comment]
By the way, clearing arrays that are found inside arrays (eg List in a Dictionary) shouldn't be required, the garbage collector handles that. Once a variable cannot be referenced to anymore, the garbage collector will clear it.
